vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java

```diff
@@ -211,6 +212,7 @@ protected boolean hasValidValue() {
     @Deprecated
     public void setDataProvider(DataProvider<T, ?> dataProvider) {
         this.dataProvider.set(dataProvider);
+        DataViewUtils.removeComponentFilterAndSortComparator(this);
```

Comment:
Would be nice to see this validated in a unit test

Reply:
Validated in `AbstractListDataViewListenerTest`